test/clojupyter/core_test.clj

```diff
@@ -60,165 +66,115 @@
                 zmq-in       (atom [])
                 zmq-out      (atom [])
                 zmq-messages (atom [])
                 nrepl-comm   (make-mock-nrepl-comm)
                 zmq-comm     (make-mock-zmq-comm zmq-in zmq-out zmq-messages)
-                handler      (configure-shell-handler states zmq-comm nrepl-comm
-                                                      socket signer)] ?form))
+                S	     {:states states :zmq-comm zmq-comm :nrepl-comm nrepl-comm
+                              :signer signer :checker checker :socket socket}
+                handler      (configure-shell-handler S)
+                S	     (assoc S :handler handler)
+                rcv	     (rcomp mzmq/parts-to-message build-message)]
+            ?form))
   (before :facts (log/set-level! :error))])
 
 (against-background
- [(before :facts (do (reset! zmq-in (list {:header
-                                           (cheshire/generate-string
-                                            {:msg_type "kernel_info_request"})
-                                      }))
+ [(before :facts (do (reset! zmq-in (list {:header (json {:msg_type "kernel_info_request"})}))
                      (doseq [_ @zmq-in]
-                       (process-event states zmq-comm socket signer handler))))]
+                       (process-event S))))]
  (fact "should be able to process kernel-info-request"
-       (get-in ((comp parse-message mzmq/parts-to-message)
-                (get @zmq-messages 0))
-               [:content :execution_state]
-               )
-       => "busy"
-       (get-in ((comp parse-message mzmq/parts-to-message)
-                (get @zmq-messages 1))
-               [:content]
-               )
-       => {:status "ok",
-           :protocol_version "5.0",
-           :implementation "clojupyter",
-           :language_info
-           {:name "clojure",
-            :version "1.8.0",
-            :mimetype "text/x-clojure",
-            :file_extension ".clj"},
-           :banner "Clojupyters-0.1.0",
-           :help_links []}
-       (get-in ((comp parse-message mzmq/parts-to-message)
-                (get @zmq-messages 2))
-               [:content :execution_state]
-               )
-       => "idle"))
+       (mapv (rcomp rcv :content #(select-keys % [:status :execution_state])) @zmq-messages)
+       => [{:execution_state "busy"}
+           {:status "ok"}
+           {:execution_state "idle"}]
+       (-> (get @zmq-messages 1) rcv (get-in [:content])
+           ((juxt :protocol_version :status :implementation
+                  (rcomp :language_info :name)
+                  (rcomp :language_info :mimetype)
+                  (rcomp :language_info :file_extension)
+                  (rcomp :language_info :version))))
+       => [messages/PROTOCOL-VERSION "ok" "clojupyter" "clojure" "text/x-clojure" ".clj"
+           (apply format "%d.%d.%d" ((juxt :major :minor :incremental) *clojure-version*))]))
```

Comment:
And I think here, we need to pull in the clojupyter version to get the kernel version, not the clojure version.

Reply:
No, I think it's correct.  The whole form is 

```
(-> (get @zmq-messages 1) rcv (get-in [:content])
    ((juxt :protocol_version :status :implementation
           (rcomp :language_info :name)
           (rcomp :language_info :mimetype)
           (rcomp :language_info :file_extension)
           (rcomp :language_info :version))))
=> [messages/PROTOCOL-VERSION "ok" "clojupyter" "clojure" "text/x-clojure" ".clj"
    (apply format "%d.%d.%d" ((juxt :major :minor :incremental) *clojure-version*))]
```
and `:protocol-version` lines up with `PROTOCOL-VERSION`, `:status` with `"ok"` and so on sequentially down the vector (`juxt` maintains order).

Reply:
Or, maybe I misunderstand: `:implementation` currently gives `"clojupyter"`, I'm pretty sure I didn't change that.  Are you saying it should be something like `"clojupyter-0.2.1-SNAPSHOT"`instead?